recipes/argon2/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class Argon2Conan(ConanFile):
+    name = "argon2"
+    license = "Apache 2.0", "CC0-1.0"
+    homepage = "https://github.com/P-H-C/phc-winner-argon2"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Argon2 password hashing library"
+    topics = ("argon2", "crypto", "password hashing")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
```

Comment:
Refactored out a duplicated code for my suggestions:

```suggestion

    @property
    def _arch(self):
        return str(self.settings.arch).replace("_", "-")
```